bundles/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/discovery/PLCDiscoveryService.java

```diff
@@ -107,44 +101,19 @@ protected void startScan() {
 
         logger.debug("Start scan for LOGO! bridge");
 
-        Enumeration<NetworkInterface> devices = null;
-        try {
-            devices = NetworkInterface.getNetworkInterfaces();
-        } catch (SocketException exception) {
-            logger.warn("LOGO! bridge discovering: {}.", exception.toString());
-        }
-
-        Set<String> addresses = new TreeSet<>();
-        while ((devices != null) && devices.hasMoreElements()) {
-            NetworkInterface device = devices.nextElement();
-            try {
-                if (!device.isUp() || device.isLoopback()) {
-                    continue;
-                }
-            } catch (SocketException exception) {
-                logger.warn("LOGO! bridge discovering: {}.", exception.toString());
-            }
-            for (InterfaceAddress iface : device.getInterfaceAddresses()) {
-                InetAddress address = iface.getAddress();
-                if (address instanceof Inet4Address) {
-                    String prefix = String.valueOf(iface.getNetworkPrefixLength());
-                    SubnetUtils utilities = new SubnetUtils(address.getHostAddress() + "/" + prefix);
-                    addresses.addAll(Arrays.asList(utilities.getInfo().getAllAddresses()));
-                }
-            }
-        }
-
+        List<InetAddress> addressesToScan = NetUtil.getFullRangeOfAddressesToScan();
```

Comment:
Previous code was ignoring interfaces that were no up and also loopback interface.
The new code does not ?

Reply:
`NetUtil.getFullRangeOfAddressesToScan()`  calls `NetUtil.getAllInterfaceAddresses()` and that method filters:

https://github.com/openhab/openhab-core/blob/389f6a34343a723442829b7d9841240adc2f2a9c/bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java#L344

Reply:
Fine